wazo_calld/plugins/calls/services.py

```diff
@@ -224,12 +224,26 @@ def originate(self, request):
             channel = self._ari.channels.get(channelId=channel_id)
 
         else:
+            error_message = None
             if 'line_id' in request['source']:
-                endpoint = user.line(request['source']['line_id']).interface()
+                requested_endpoint = user.line(request['source']['line_id'])
+                if not requested_endpoint.is_online(self._ari):
+                    error_message = "Could not dial caller's requested line"
+                endpoint = requested_endpoint.interface()
+
             elif request['source']['all_lines']:
                 endpoint = f"local/{source_user}@usersharedlines"
             else:
-                endpoint = user.main_line().interface()
+                requested_endpoint = user.main_line()
+                if not requested_endpoint.is_online(self._ari):
+                    error_message = "Could not dial caller's main line"
+                endpoint = requested_endpoint.interface()
+
+            if error_message is not None:
+                details = {
+                    'caller_extension': endpoint,
+                }
+                raise CallCreationError(error_message, details=details)
```

Comment:
`CallCreationError` / `error_id="call_creation"` seems pretty generic to me. Could be worthwhile to create a more specific error, e.g. `call-origin-endpoint-unavailable` . This would allow client apps to correctly detect  and perhaps take actions to fix the problem(e.g. sending a wake up request to some other network entity to make the endpoint available).

Reply:
Will do :+1:

Reply:
Added `call-origin-unavailable-error`